src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberCommon.java

```diff
@@ -37,14 +37,29 @@
     
     private static final String DEFAULT_DATE_FORMAT_PATTERN = "yyyy-MM-dd";
     
-    public static VersionNumberBuildInfo incBuild(Run build, Run prevBuild, boolean skipFailedBuilds) {
+    // Pattern:   ${VAR_NAME} or $VAR_NAME
+    public static final String ENV_VAR_PATTERN = "^(?:\\$\\{(\\w+)\\})|(?:\\$(\\w+))$";
+    
+    public static VersionNumberBuildInfo incBuild(Run build, EnvVars environmentVariables,
+    		Run prevBuild, boolean skipFailedBuilds, String overrideBuildsAllTime) {
         int buildsToday = 1;
         int buildsThisWeek = 1;
         int buildsThisMonth = 1;
         int buildsThisYear = 1;
         int buildsAllTime = 1;
         // this is what we add to the previous version number to get builds today / this week / this month / this year / all time
         int buildInc = 1;
+
+        /*
+         * If we're overriding builds all time, figure it out here.  If we don't
+         * it will be business as usual and we'll increment as needed.
+         * 
+         * It's necessary to do this here in case you want to override the build
+         * number on the very first build.
+         */
```

Comment:
To be honest, I do not understand why you split setting the value of `buildsAllTime` to two different places? (Here and 60 lines further down.)
Why don't you just do it 60 lines further down?